src/DatabaseSchemaManager.php

```diff
@@ -0,0 +1,251 @@
+                        FROM sqlite_master m
+                        LEFT OUTER JOIN pragma_table_info((m.name)) p
+                        ON m.name <> p.name
+                        WHERE m.type = 'table'
+                        ORDER BY table_name, col_id");
+                    break;
+
+                default:
+                    $rawColumns = DB::select("SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE table_schema = '".config('database.connections.'.$connection.'.database')."' ORDER BY table_name, ordinal_position");
+
+            }
+
+            $this->schema[$connection] = self::mapColumns($rawColumns);
+        }
+    }
+
+    /**
+     * Maps the columns from raw db values into an usable array.
+     *
+     * @param  mixed  $rawColumns
+     * @return array
+     */
+    protected static function mapColumns($rawColumns)
+    {
+        switch (DB::getPdo()->getAttribute(\PDO::ATTR_DRIVER_NAME)) {
+            case 'sqlite' :
+                $mappedColumns = collect($rawColumns)->mapToGroups(function ($item, $key) {
+                    return [$item->TABLE_NAME => [
+                        'name' => $item->COLUMN_NAME,
+                        'is_nullable' => (bool) $item->IS_NULLABLE === true ? false : true, // sqlite uses "is not nullable" instead of "is nullable"
+                        'is_primary' => $item->COLUMN_PK >= 1 ? true : false,
+                        'default' => $item->COLUMN_DEFAULT ?? null,
+                    ]];
+                });
+                break;
+
+            default:
+                $mappedColumns = collect($rawColumns)->mapToGroups(function ($item, $key) {
+                    return [$item->TABLE_NAME => [
+                        'name' => $item->COLUMN_NAME,
+                        'is_nullable' => $item->IS_NULLABLE === 'YES' ? true : false,
+                        'is_primary' => $item->COLUMN_KEY === 'PRI' ? true : false,
+                        'default' => $item->COLUMN_DEFAULT ?? null,
+                    ]];
+                });
+
+        }
+
+        return $mappedColumns->toArray();
+    }
+
+    /**
+     * Check if the column exists in the schema.
+     *
+     * @param  string  $connetionName
+     * @param  string  $tableName
+     * @param  string  $columnName
+     * @return bool
+     */
+    public function hasColumn($connectionName, $tableName, $columnName)
+    {
+        if (! $this->ensureSchemaExistence($connectionName) || ! $this->ensureTableExistence($connectionName, $tableName)) {
+            return false;
+        }
+
+        return in_array($columnName, array_column($this->schema[$connectionName][$tableName], 'name'));
+    }
+
+    /**
+     * Check if the column is nullable in database.
+     *
+     * @param  string  $connetionName
+     * @param  string  $tableName
+     * @param  string  $columnName
+     * @return bool
+     */
+    public function isColumnNullable($connectionName, $tableName, $columnName)
```

Comment:
I think `columnIsNullable()` would be a better name here, like we have `columnHasDefault()`. It just reads more natural to me since it's about something else... If it were `isNullable` I'd agree with you, because the main object would be the subject of that sentence "field->isNullable", but in this case, "field->isColumnNullable" reads a bit backwards.

So I'd prefer `columnIsNullable()` - what do you think?